cloud-platform/app-deploy.md

```diff
@@ -0,0 +1,332 @@
+
+For the value of the `image:` key, you will see the Repository URI, which has been provided by ECR.
+
+If you are using the file as a template for your own application, replace the value of `image:` with the Repository URI of your application, found in ECR.
+
+### Service
+
+Service files are used to specify port and protocol information for your application and are also used to bundle together the set of pods created by the deployment.
+
+See the contents of the `service.yml` file below:
+
+```Yaml
+kind: Service
+apiVersion: v1
+metadata:
+  name: cp-app-demo-svc
+  labels:
+    app: cp-app-demo-svc
+spec:
+  ports:
+  - port: 80
+    name: http
+    targetPort: 80
+  selector:
+    app: cp-demo-app
+```
+
+If you are using the file as a template for your own application, replace the `cp-demo-app-svc` tags with ones suited to your application.
+
+Also, ensure the `selector:` app tag is the same as specified in the `deployment.yml`.
+
+### Ingress
+
+Ingress files are to use to define external access to the application.
+
+See the contents of the `ingress.yml` file below:
+
+```Yaml
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: cp-demo-app-ing
+spec:
+  rules:
+  - host: cp-demo-app.integration.dsd.io
+    http:
+      paths:
+      - path: /
+        backend:
+          serviceName: cp-demo-app-svc
+          servicePort: 80
+```
+
+If you are using the file as a template for your own application, replace the `cp-demo-app-ing` tag with one suited to your application.
+
+Also, ensure the `serviceName:` tag is the same as specified in the `service.yml`.
+
+## Configuring secrets
+
+The following is an example of encoding (configuring) aws access-key credentials in your deployment.
+(See https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables
+for detailed information regarding providing base64 values in secret objects to Kuberenetes pods)
+
+Create your AWS Credentials access key (making a note of the aws_access_key_id and aws_secret_access_key)
+
+put the aws key secret and id into a file (in this example demo-v9.k8s) as follows ('thekey' referencing the aws_access_key_id
+and 'thesecret' referencing the aws_secret_access_key):
+
+```
+apiVersion: v1
+
+kind: Secret
+
+metadata:
+
+  name: demosecret-v9
+
+type: Opaque
+
+data:
+
+  thekey: AKIAFTKSAW15HJLOGD
+
+  thesecret: g8hjpmhvgfhk4547gfdshhjj
+
+```
+
+create secret generic secret from that file:
+```$kubectl -n demo-app create secret generic demosecret-v9 --from-file=demosecret-v9=demo-v9.k8s```
+
+'demo-app' being the namespace, 'demosecret-v9' referencing the 'name' in demo-v9.k8s and referencing the file 'demo-v9.k8s'
+this will give the following screen output
+
+
+```
+secret "demosecret-v9" created
+```
+To view the output of your secret:
+
+```
+$kubectl -o json -n demo-app get secret demosecret-v9
```

Comment:
You can use `kubectl` to get just the value of an item in a `Secret` object on its own, which is probably a bit easier to understand than the full resource output:

```
$ kubectl get secret demosecret -n demo-app -o jsonpath="{.data.aws_secret_key}"
QUtJQUZUS1NBVzE1SEpMT0dE
```

That can then be piped directly into `base64` to get the plaintext value:

```
$ kubectl get secret demosecret -n demo-app -o jsonpath="{.data.aws_secret_key}" | base64 --decode
AKIAFTKSAW15HJLOGD
```